### PR TITLE
Add nx-cugraph Landing Page

### PR DESCRIPTION
**Pull request by nv-rliu**
Closes https://github.com/rapidsai/graph_dl/issues/592

Continuation and replacement of #415 (Creating separate PR so I can directly add changes as needed. Preserved commits from @exactlyallan )

Adding the `nx-cugraph` landing page for RAPIDS

### CODE REVIEW

File: content/_partners/networkx/index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+name: Networkx
```
Comment (bdice) [inline content/_partners/networkx/index.md:2]:
```suggestion
name: NetworkX
```

File: content/nx-cugraph/index.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+title: nxCuGraph
```
Comment (bdice) [inline content/nx-cugraph/index.md:2]:
```suggestion
title: nx-cugraph
```

File: content/nx-cugraph/index.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+title: nxCuGraph 
+description: GPU Accelerated NetworkX Backends
```
Comment (bdice) [inline content/nx-cugraph/index.md:3]:
```suggestion
description: GPU Accelerated NetworkX Backend
```

File: layouts/page/nx-cugraph.html [hunk 4]
```diff
@@ -0,0 +1,325 @@
+{{- define "main" -}}
+
+<!-- Hero Partial -->
+{{ partial "hero.html" (dict
+"text" "GPU Accelerated NetworkX Backends"
```
Comment (bdice) [inline layouts/page/nx-cugraph.html]:
```suggestion
"text" "GPU Accelerated NetworkX Backend"
```

File: layouts/page/nx-cugraph.html [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+<section class="row section mb-6">
+  <div class="col">
+    <div class="container">
+
+      <h1 class="text-center mb-6">Try Zero Code Change Accelerated NetworkX</h1>
+
+        <div class="row">
+          <div class="col-md-4 py-3">
+            <h2><i class="fa-light fa-gauge-high"></i> Zero Code Change Acceleration </h2>
+            <p >Write your code with the full ease of Networkx. It's as easy as setting a single environment variable to begin accelerating your code on the GPU. </p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html]:
```suggestion
            <p >Write your code with the full ease of NetworkX. It's as easy as setting a single environment variable to begin accelerating your code on the GPU. </p>
```

File: layouts/page/nx-cugraph.html [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+<div class="bottom-slant-container">
+  <div class="bottom-slant-down bg-white"></div>
+</div>
+
+<section class="row section padding-top-slant padding-bottom-slant">
+  <div class="col">
+    <div class="container">
+
+      <h1 class="mb-4"> Bringing GPU Accelerated graph analytics <br> to every NetworkX User </h1>
+      <p class="subheading mb-5">Bring blazing fast performance to your existing Graph workflows.</p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html:55]:
```suggestion
      <p class="subheading mb-5">Bring blazing fast performance to your graph workflows.</p>
```

File: layouts/page/nx-cugraph.html [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+          
+          <p>From the shell, set an environment variable: </p>
+          <code class="code-sample mt-3">
+            NX_CUGRAPH_AUTOCONFIG=True<br>
+            python my_networkx_script.py
+          </code>
+
+          <br>
+
+          <p>From a notebook, use %env before the networkx import:</p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html:73]:
```suggestion
          <p>From a notebook, use %env before importing NetworkX:</p>
```

File: layouts/page/nx-cugraph.html [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+      <h1 class="mt-10 mb-4"> How It Works </h1>
+
+      <div class="row">
+        <div class="col-md-6 py-3">
+          <h2><i class="fa-regular fa-light fa-gear-code"></i> Under the Hood </h2>
+          <p>NetworkX has the ability to dispatch function calls to separately-installed third-party backends. These NetworkX backends let users experience improved performance and/or additional functionality without changing their NetworkX Python code. Examples include backends that provide algorithm acceleration using GPUs, parallel processing, graph database integration, and more.</p>
+          
+          <br>
+          
+          <p>While NetworkX is a pure-Python implementation, backends may be written to use other libraries and even specialized hardware. As such, `nx-cugraph` is a NetworkX backend that uses RAPIDS cuGraph and NVIDIA GPUs to significantly improve NetworkX performance.</p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html:120]: Code formatting with backticks is Markdown syntax and is not supported in HTML. Let's use plain text here instead.
```suggestion
          <p>While NetworkX is a pure-Python implementation, backends may be written to use other libraries and even specialized hardware. As such, nx-cugraph is a NetworkX backend that uses RAPIDS cuGraph and NVIDIA GPUs to significantly improve NetworkX performance.</p>
```

File: layouts/page/nx-cugraph.html [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+        <div class="col-md-6 py-3">
+          <h2><i class="fa-light fa-ballot-check"></i> Execution Flow </h2>
+          {{ with .Resources.GetMatch "nxcg-execution-diagram.jpg" }}
+          <img src="{{ .RelPermalink }}">
+          {{ end }}
+        </div>
+      </div>
+
+      <h1 class="mt-10 mb-4"> Try nx-cugraph Yourself </h1>
+      <p class="subheading  mb-5">Find where to get more resources and connect with developers.</p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html:173]:
```suggestion
      <p class="subheading  mb-5">Get more resources and connect with developers.</p>
```

File: layouts/page/nx-cugraph.html [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+          <h2><i class="fa-light fa-bolt-lightning"></i> Try Now</h2>
+          <p>See how nx-cugraph can speedup your graph workflows in a free <a href="https://nvda.ws/4drM4re" target="_blank">GPU-enabled Colab notebook environment</a> using your Google account.</p>
+         
+          <br>
+         
+          <p>Additionally, nx-cugraph is now included in the RAPIDS metapackage. Visit the <a href="https://rapids.ai/#quick-start">RAPIDS Quick Start</a> to get started with RAPIDS on your favorite platform.</p>
+        </div>
+        <div class="col-md-6 py-3">
+          <h2><i class="fa-light fa-comment-code"></i> Learn More </h2>
+          </p>Now nx-cugraph is generally available (GA) and ready for wide use. You can learn more from the <a href="https://developer.nvidia.com/blog/networkx-introduces-zero-code-change-acceleration-using-nvidia-cugraph/" target="_blank"> release blog</a> and <a href="https://networkx.org/documentation/stable/tutorial.html#using-networkx-backends" target="_blank"> nx-cugraph documentation</a>. </p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html:186]: I expected the second link to go to the nx-cugraph docs, not the NetworkX docs on using backends.
```suggestion
          </p>Now nx-cugraph is generally available (GA) and ready for wide use. You can learn more from the <a href="https://developer.nvidia.com/blog/networkx-introduces-zero-code-change-acceleration-using-nvidia-cugraph/" target="_blank"> release blog</a> and <a href="https://docs.rapids.ai/api/cugraph/stable/nx_cugraph/nx_cugraph/" target="_blank"> nx-cugraph documentation</a>. </p>
```

File: layouts/page/nx-cugraph.html [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+          
+          <br>
+          
+          Interested to see nx-cugraph used in an end-to-end workflow - try this <a href="https://colab.research.google.com/github/rapidsai-community/showcase/blob/HEAD/accelerated_data_processing_examples/nxcg_wikipedia_e2e.ipynb" target="_blank"> SciPy conference demo notebook</a>. Want to contribute or share feedback - reach out via <a href="https://github.com/rapidsai/nx-cugraph" target="_blank">GitHub</a>.</p>
```
Comment (bdice) [inline layouts/page/nx-cugraph.html:190]:
```suggestion
          Interested in seeing nx-cugraph used in an end-to-end workflow? Try this <a href="https://colab.research.google.com/github/rapidsai-community/showcase/blob/HEAD/accelerated_data_processing_examples/nxcg_wikipedia_e2e.ipynb" target="_blank"> SciPy conference demo notebook</a>. Want to contribute or share feedback? Reach out via <a href="https://github.com/rapidsai/nx-cugraph" target="_blank">GitHub</a>.</p>
```

File: layouts/partials/footer.html [hunk 12]
```diff
@@ -38,7 +38,11 @@
         <div class="footer-row text-center">
             <a class="footer-item" href="/cudf-pandas"><i class="fa-light fa-gauge-circle-bolt"></i> cuDF pandas</a>
             <span class="footer-item"> | </span>
+            <a class="footer-item" href="/cuvs"><i class="fa-light fa-chart-scatter-3d"></i> cuVS</a>
+            <span class="footer-item"> | </span>
             <a class="footer-item" href="/polars-gpu-engine"><i class="fa-light fa-igloo"></i> Polars GPU Engine</a>
+            <span class="footer-item"> | </span>
+            <a class="footer-item" href="/nx-cugraph"><i class="fa-sharp fa-light fa-chart-network"></i> nx-cuGraph</a>
```
Comment (bdice) [inline layouts/partials/footer.html:45]: I'm questioning the footer layout here. We have "landing pages" for cuvs, cudf.pandas, cudf-polars, and nx-cugraph. But we don't have such pages for every library. Should these be in some kind of top-level menu instead of a footer? This proposal is out of scope for this PR... but probably something we should consider.

Comment (exactlyallan) [reply layouts/partials/footer.html:45]: Agree its a bit awkward - with the docs merge, itll make sense to either remove these pages or move them into the "docs" section which would mean each library has an overview page.

File: layouts/page/nx-cugraph.html (hunk 10 above)
Comment (exactlyallan) [reply layouts/page/nx-cugraph.html:186]: @nv-rliu is that correct?

**Comment by nv-rliu** [comment]
@bdice I updated the PR with just the suggestion here https://github.com/rapidsai/rapids.ai/pull/419#discussion_r1828127860

The typos were already resolved but thank you for pointing them out and giving everything another look. Re-requesting review!

**Comment by rlratzel** [comment]
Looks good.  This isn't a requirement, but I noticed the "Test Drive cuDF" blurb with a link to the cudf colab notebook.  Do we want the same for cugraph? "Test Drive NetworkX with cuGraph"?  I don't have a strong opinion either way.

**Review by rlratzel — APPROVED** [review]

**Comment by nv-rliu** [comment]
> Looks good. This isn't a requirement, but I noticed the "Test Drive cuDF" blurb with a link to the cudf colab notebook. Do we want the same for cugraph? "Test Drive NetworkX with cuGraph"? I don't have a strong opinion either way.

Hey Rick, where is this section?

**Comment by yonn15** [comment]
Header has a descriptor where the others don't:   NetworkX, Supercharged With GPU Acceleration
Suggest this alt: NetworkX Accelerated by cuGraph OR NetworkX Supercharged by cuGraph

This box should line up with NetworkX benchmark below it.

Can the logo be a tad bigger to resize similar to the other logos?

File: layouts/index.html [hunk 13]
```diff
@@ -90,18 +90,25 @@ <h2><i class="fa-light fa-chart-network"></i> Open Source Ecosystem </h2>
...
 
             <div class="row has-background-rapids-image has-box-shadow">
-                <div class="col-md-6 py-5">
+                <div class="col-md-4 py-5">
                     <h2 class="text-white"><i class="fa-light fa-gauge-circle-bolt"></i> Pandas Accelerator Mode for cuDF </h2>
                     <p class="text-white">Use cuDF pandas Accelerator Mode to speed up pandas workflows with zero code change.
                         <a href="/cudf-pandas"> Learn More on the Accelerator Mode Page <i class="fa-solid fa-arrow-up-right"></i></a>
                     </p>
-
+
```
Comment (bdice) [inline layouts/index.html:98]: Remove extraneous whitespace from this PR.
```suggestion

```

File: layouts/index.html [hunk 14]
```diff
@@ -340,16 +347,26 @@ <h2>Open Source</h2>
             <p>RAPIDS would not be possible without the collaboration of these important open source projects. Click on a logo to
               learn more: </p>
             <div class="d-flex flex-wrap justify-content-between mt-5">
-              {{ $row := slice "apache-arrow" "dask" "nuclio" "numba" "scikit-learn" "xgboost" }}
+              {{ $row := slice "apache-arrow" "dask" "nuclio" "networkx" "numba" "scikit-learn" "xgboost" }}
```
Comment (bdice) [inline layouts/index.html:350]: These are alphabetical. Let’s match the existing style.
```suggestion
              {{ $row := slice "apache-arrow" "dask" "networkx" "nuclio" "numba" "scikit-learn" "xgboost" }}
```

File: layouts/index.html [hunk 15]
```diff
@@ -340,16 +347,26 @@ <h2>Open Source</h2>
...
-              {{ $resized_image := ($sm_logo | default $default_logo).Resize "100x" }}
-              <a target="_blank" href="{{ .Params.github_repo_url | default .Params.github_org_url | default .Params.homepage_url }}">
-                <img src="{{ $resized_image.RelPermalink }}" alt="{{ .Params.name }} Logo">
-              </a>
-              {{ end }}
+                {{ with site.GetPage (printf "_partners/%s" . ) }}
+                  {{ $sm_logo := .Resources.Get "logo_400x.png" }}
+                  {{ $default_logo := .Resources.Get "logo.png" }}
+
+                  <!-- check to render logo differently. NetworkX logo is too small normally -->
```
Comment (bdice) [inline layouts/index.html:356]: Can we get a logo that is the appropriate size instead of adding conditional logic here?

Comment (nv-rliu) [reply layouts/index.html:355]: So all the logos live inside `content/_partners/<partner_name>` in their high-resolution format. However, `layouts/index.html` will resize any image to a set width value (100px) and the NetworkX logo has a different aspect ratio than the other logos making it appear a lot smaller. 

I'm not sure what another workaround would be without modifying too much of the existing code

**Review by bdice — APPROVED** [review]

**Comment by bdice** [comment]
@exactlyallan Please merge once you approve. I would appreciate your feedback on this thread: https://github.com/rapidsai/rapids.ai/pull/419#discussion_r1829605768

**Comment by exactlyallan** [comment]
> @exactlyallan Please merge once you approve. I would appreciate your feedback on this thread: [#419 (comment)](https://github.com/rapidsai/rapids.ai/pull/419#discussion_r1829605768)

Updated the ratio so fixed.
